server/main.lua

```diff
@@ -36,6 +36,14 @@ RegisterNetEvent('qb-vehiclekeys:server:AcquireVehicleKeys', function(plate)
     GiveKeys(src, plate)
 end)
 
+RegisterNetEvent('qb-vehiclekeys:server:breakLockpick', function(itemName)
+    local Player = QBCore.Functions.GetPlayer(source)
+    if Player then
+        TriggerClientEvent("inventory:client:ItemBox", source, QBCore.Shared.Items[itemName], "remove")
+        Player.Functions.RemoveItem(itemName, 1)
+    end
```

Comment:
```suggestion
    if not Player then return end
    if not (itemName == "lockpick" or itemName == "advancedlockpick") then return end
    if Player.Functions.RemoveItem(itemName, 1) then
            TriggerClientEvent("inventory:client:ItemBox", source, QBCore.Shared.Items[itemName], "remove")
    end
```

Cleaner code and not exploitable.